include/blas_quda.h

```diff
@@ -28,261 +28,308 @@ namespace quda {
       for (auto i = 0u; i < x.size(); i++) x[i].zero();
     }
 
-    inline void copy(ColorSpinorField &dst, const ColorSpinorField &src) { dst.copy(src); }
+    inline void copy(cvector_ref<ColorSpinorField> &dst, cvector_ref<const ColorSpinorField> &src)
+    {
+      for (auto i = 0u; i < src.size(); i++) { dst[i].copy(src[i]); }
+    }
 
     /**
        @brief Apply the operation y = a * x
-       @param[in] a scalar multiplier
-       @param[in] x input vector
-       @param[out] y output vector
+       @param[in] a scalar multiplier set
+       @param[in] x input vector set
+       @param[out] y output vector set
     */
-    void axy(double a, const ColorSpinorField &x, ColorSpinorField &y);
+    void axy(cvector<Complex> &a, cvector_ref<const ColorSpinorField> &x, cvector_ref<ColorSpinorField> &y);
 
     /**
        @brief Apply the rescale operation x = a * x
-       @param[in] a scalar multiplier
-       @param[in] x input vector
+       @param[in] a scalar multiplier set
+       @param[in] x set of vectors
     */
-    inline void ax(double a, ColorSpinorField &x) { axy(a, x, x); }
+    inline void ax(cvector<Complex> &a, cvector_ref<ColorSpinorField> &x) { axy(a, x, x); }
```

Comment:
`cvector<double>` (again)?

Reply:
As above an intentional change to support complex scalar multiplications.